nerfstudio/engine/trainer.py

```diff
@@ -104,39 +100,39 @@ class Trainer:
     optimizers: Optimizers
     callbacks: List[TrainingCallback]
 
-    def __init__(self, config: TrainerConfig, local_rank: int = 0, world_size: int = 1):
+    def __init__(self, config: TrainerConfig, local_rank: int = 0, world_size: int = 1) -> None:
         self.config = config
         self.local_rank = local_rank
         self.world_size = world_size
-        self.device = "cpu" if world_size == 0 else f"cuda:{local_rank}"
-        self.mixed_precision = self.config.mixed_precision
+        self.device: str = "cpu" if world_size == 0 else f"cuda:{local_rank}"
```

Comment:
Maybe this should be `Union[torch.device, str]`

Reply:
Addressed in [71cadf2](https://github.com/nerfstudio-project/nerfstudio/pull/1509/commits/71cadf2d0ebd7815de1590340174268ff954e349)